tcms/testcases/views.py

```diff
@@ -603,120 +604,29 @@ def printable(request,  # pylint: disable=missing-permission-required
-    # done by simple_history. Should we remove it ???
-    # Modify the contents
-    fields = ['summary',
-              'case_status',
-              'category',
-              'priority',
-              'notes',
-              'text',
-              'is_automated',
-              'script',
-              'arguments',
-              'extra_link',
-              'requirement']
-
-    for field in fields:
-        if getattr(test_case, field) != tc_form.cleaned_data[field]:
-            setattr(test_case, field, tc_form.cleaned_data[field])
-    try:
-        if test_case.default_tester != tc_form.cleaned_data['default_tester']:
-            test_case.default_tester = tc_form.cleaned_data['default_tester']
-    except ObjectDoesNotExist:
-        pass
-
-    test_case.save()
-
-
-@permission_required('testcases.change_testcase')
-def edit(request, case_id):
-    """Edit case detail"""
-    try:
-        test_case = TestCase.objects.select_related().get(case_id=case_id)
-    except ObjectDoesNotExist:
-        raise Http404
-
-    test_plan = plan_from_request_or_none(request)
+    model = TestCase
+    template_name = 'testcases/mutable.html'
+    form_class = EditCaseForm
 
-    if request.method == "POST":
-        form = NewCaseForm(request.POST)
-        if request.POST.get('product'):
-            form.populate(product_id=request.POST['product'])
-        elif test_plan:
-            form.populate(product_id=test_plan.product_id)
+    def get_form(self, form_class=None):
+        form = super().get_form()
+        if self.request.POST.get('product'):
+            form.populate(product_id=self.request.POST['product'])
         else:
-            form.populate()
-
-        n_form = CaseNotifyForm(request.POST)
-
-        if form.is_valid() and n_form.is_valid():
-            update_testcase(request, test_case, form)
-            update_case_email_settings(test_case, n_form)
-
-            return HttpResponseRedirect(
-                reverse('testcases-get', args=[case_id, ])
-            )
-
-    else:
-        # Notification form initial
-        n_form = CaseNotifyForm(initial={
-            'notify_on_case_update': test_case.emailing.notify_on_case_update,
-            'notify_on_case_delete': test_case.emailing.notify_on_case_delete,
-            'author': test_case.emailing.auto_to_case_author,
-            'default_tester_of_case': test_case.emailing.auto_to_case_tester,
-            'managers_of_runs': test_case.emailing.auto_to_run_manager,
-            'default_testers_of_runs': test_case.emailing.auto_to_run_tester,
-            'assignees_of_case_runs': test_case.emailing.auto_to_case_run_assignee,
-            'cc_list': MultipleEmailField.delimiter.join(
-                test_case.emailing.get_cc_list()),
-        })
-
-        components = []
-        for component in test_case.component.all():
-            components.append(component.pk)
+            form.populate(product_id=self.object.category.product_id)
```

Comment:
This one is not needed, it should still be like before.

The original logic was to filter categories if product was selected. That would ensure that only Categories related to the currently selected product are considered as valid. This is done by filtering the queryset of the form field to limit the available choices.

B/c these choices are also controlled on the FE there was this mechanism on the BE as well. 


Also I am not even sure if we need this entire piece of code around .populate when editting but will look into it more.